syntax = "proto3";
package template;





/*
DetailCollectorTemplate请求
*/
message DetailCollectorTemplateRequest {
    /*
    实例ID
    */
    string instanceId = 1 ;
}

/*
DetailCollectorTemplate返回
*/
message DetailCollectorTemplateResponse {
message CollectorMetrics {
message TagDefine {
    /*
    名字
    */
    string name = 1 ;
    /*
    默认值
    */
    string default = 2 ;
    /*
    是否只读
    */
    bool readOnly = 3 ;
    /*
    描述
    */
    string description = 4 ;
    /*
    值类型
    */
    string valueType = 5 ;
}
        
    
    /*
    指标名
    */
    string name = 1 ;
    /*
    单位
    */
    string unit = 2 ;
    /*
    指标数据类型
    */
    string dataType = 3 ;
    /*
    指标类型
    */
    string metricType = 4 ;
    /*
    帮助
    */
    string help = 5 ;
    /*
    指标key
    */
    string key = 6 ;
    /*
    维度定义
    */
    repeated TagDefine tagDefine = 7 ;
}
        
message MetricNames {
    /*
    指标id
    */
    string metricId = 1 ;
    /*
    指标名
    */
    string metricName = 2 ;
    /*
    指标是否只用于中间计算，不上报，默认为false
    */
    bool calculateOnly = 3 ;
}
        
message Params {
    /*
    key
    */
    string key = 1 ;
    /*
    value
    */
    string value = 2 ;
    /*
    参数类型
    */
    string paramType = 3 ;
}
        
message ProcessSteps {
    /*
    名称
    */
    string name = 1 ;
    /*
    动作
    */
    string action = 2 ;
    /*
    参数,json格式字符串
    */
    string params = 3 ;
    /*
    上一级处理步骤名称列表
    */
    repeated string parents = 4 ;
}
        
message MetricProperties {
    /*
    关联指标instanceId
    */
    string metricId = 1 ;
    /*
    指标是否只用于中间计算，不上报，默认为false
    */
    bool calculateOnly = 2 ;
}
        
    
    /*
    关联指标
    */
    repeated CollectorMetrics collectorMetrics = 1 ;
    /*
    实例ID
    */
    string instanceId = 2 ;
    /*
    名称
    */
    string name = 3 ;
    /*
    采集间隔
    */
    int32 interval = 4 ;
    /*
    指标名映射列表(deprecated)
    */
    repeated MetricNames metricNames = 5 ;
    /*
    采集模型id
    */
    string objectId = 6 ;
    /*
    采集参数
    */
    repeated Params params = 7 ;
    /*
    结构体数组/关系展开字段
    */
    string expand = 8 ;
    /*
    采集类型
    */
    string type = 9 ;
    /*
    指标处理流程
    */
    repeated ProcessSteps processSteps = 10 ;
    /*
    所属Agent
    */
    string agentType = 11 ;
    /*
    模版配置yaml
    */
    string configYaml = 12 ;
    /*
    关联指标属性
    */
    repeated MetricProperties metricProperties = 13 ;
    /*
    关联日志id
    */
    string logId = 14 ;
}

/*
DetailCollectorTemplateApi返回
*/
message DetailCollectorTemplateResponseWrapper {
    /*
    返回码
    */
    int32 code = 1 ;
    /*
    返回码解释
    */
    string codeExplain = 2 ;
    /*
    错误详情
    */
    string error = 3 ;
    /*
    返回数据
    */
    template.DetailCollectorTemplateResponse data = 4 ;
}

